app/services/delete_user_in_auth0.rb

```diff
@@ -0,0 +1,20 @@
+class DeleteUserInAuth0
+  attr_accessor :user
+
+  def initialize(user:)
+    self.user = user
+  end
+
+  def call
```

Comment:
As I mentioned in another comment, this action does not perform the changes transactionally or in a safe order -- we delete the user from Auth0, we don't check the result of that, and then we delete the `auth_id` from our database, again without checking the result.

This could produce an end state where the user is deleted from Auth0 but not in our DB, or via versa. Would it be worth applying the same transaction pattern as we use when creating the user, i.e.:

- Set `auth_id` to `nil`
- Delete the user from Auth0
    - if this succeeds, commit the transaction
    - otherwise, rollback

Reply:
Quite right. I've now amended the refactoring of the deactivation commit to include this. 

The only change I made from the previous pattern was to update the local user record _after_ attempting the external request, since we need the `auth_id` to be present for the request. I think that passing an object to `DeleteAuth0User` is preferable to making a variable to store the `auth_id` before hand and passing along a string. I'm not sure there is any downside to putting it after the `begin`?

Reply:
In this case, I don't see any downside to updating the `auth_id` to nil in one step.